packages/matchbox/src/components/Modal/Modal.js

```diff
@@ -1,88 +1,144 @@
-import React, { Component } from 'react';
+import React, { useRef } from 'react';
+import { Transition } from 'react-transition-group';
+import FocusLock from 'react-focus-lock';
+import styled from 'styled-components';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
 import { Close } from '@sparkpost/matchbox-icons';
+import { ScreenReaderOnly } from '../ScreenReaderOnly';
 import { WindowEvent } from '../WindowEvent';
 import { Button } from '../Button';
-import { Grid } from '../Grid';
-import Content from './Content';
+import { Portal } from '../Portal';
 import { onKey } from '../../helpers/keyEvents';
-import styles from './Modal.module.scss';
+import { base, focusLock, wrapper, content, contentAnimation, closeButton } from './styles';
 
-class Modal extends Component {
-  static displayName = 'Modal';
+const StyledBase = styled('div')`
+  ${base}
+`;
 
-  static propTypes = {
-    /**
-     * Controlled open state of the modal
-     */
-    open: PropTypes.bool,
+const StyledWrapper = styled('div')`
+  ${wrapper}
+`;
 
-    /**
-     * An optional function that is called on key down 'Escape' and on click outside modal content
-     */
-    onClose: PropTypes.func,
+const StyledCloseButton = styled(Button)`
+  ${closeButton}
+`;
 
-    /**
-     * Modal content
-     */
-    children: PropTypes.node,
+const StyledFocusLock = styled(FocusLock)`
+  ${focusLock}
+`;
 
-    showCloseButton: PropTypes.bool
+const StyledContent = styled('div')`
+  ${content}
+  ${contentAnimation}
+`;
+
+function Modal(props) {
+  const { onClose, children, containerId, className, showCloseButton, ...rest } = props;
+  let container = useRef(null);
+  let content = useRef(null);
+
+  const handleKeydown = e => {
+    const { open } = props;
+
+    if (open && onClose) {
+      onKey('escape', onClose)(e);
+    }
   };
 
-  handleOutsideClick = (e) => {
-    const { open, onClose } = this.props;
-    const isOutside = this.content && !this.content.contains(e.target) && this.container && this.container.contains(e.target);
+  const handleOutsideClick = e => {
+    const { open } = props;
+    const isOutside =
+      content && !content.contains(e.target) && container && container.contains(e.target);
 
     if (open && isOutside && onClose) {
       onClose(e);
     }
-  }
+  };
 
-  handleKeyDown = (e) => {
-    const { onClose, open } = this.props;
+  return (
+    <Portal containerId={containerId}>
+      <StyledBase
+        {...rest}
+        className={className}
+        onClose={onClose}
+        ref={el => (container = el)}
```

Comment:
I think you could just do `ref={container}` here with `useRef`

Reply:
Ohhhhh, so because you used `let` with `useRef`, callback refs work and you don't have to use `ref.current` to access it. Wish this was documented better, but it seems doing `const ref = useRef()` changes the behavior of the hook.

Reply:
I was actually mimicking your original implementation in this component. Works well.